### PR TITLE
Add support, tests and documentation for Gentoo

### PR DESCRIPTION
**Pull request by optiz0r**

### CODE REVIEW

File: spec/classes/python_spec.rb [hunk 1]
```diff
@@ -272,4 +272,75 @@
...
     end
   end
+
+  context "on a Gentoo OS" do
+    let :facts do
+      {
+        :id => 'root',
+        :kernel => 'Linux',
+        :osfamily => 'Gentoo',
+        :operatingsystem => 'Gentoo',
```
Comment (ghoneycutt) [inline spec/classes/python_spec.rb]: you must specify `:lsbdistcodename` here for TravisCI to be happy.

https://travis-ci.org/stankevich/puppet-python/jobs/166102244

**Comment by ghoneycutt** [comment]
Thanks for adding Gentoo support. The code looks good and once the tests are passing, this should be able to be merged.

**Comment by optiz0r** [comment]
Thanks @ghoneycutt, I saw that it had failed but didn't yet find the time to investigate, thanks for pointing me directly at the fix. I've pushed this change (lsbdistcodename shows up as "n/a" for me on my system, so that's what I've used for the test). Looks like travis-ci timed out on all the ruby-2.3.1 builds, but working for all previous releases. I don't think that's a result of my code changes? The output seems to stop while running the Debian tests.

**Comment by ghoneycutt** [comment]
Yeah, that's not related to this. Could you please squash your commits and I'll do a merge and release tomorrow.

**Comment by optiz0r** [comment]
@ghoneycutt squashed and hopefully ready to go!

**Comment by ghoneycutt** [comment]
Thank you!

Released in 1.18.0

**Pull request merged**